internal/logger/logger.go

```
@@ -0,0 +1,56 @@
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+	}
+
+	// Log to console and file
+	consoleCore := zapcore.NewCore(zapcore.NewConsoleEncoder(encoderConfig), zapcore.AddSync(os.Stdout), zap.DebugLevel)
+	fileCore := zapcore.NewCore(zapcore.NewJSONEncoder(encoderConfig), getLogFileWriter("logs/app.log"), zap.InfoLevel)
+
+	// Combine cores
+	core := zapcore.NewTee(consoleCore, fileCore)
+	log = zap.New(core, zap.AddCaller(), zap.AddStacktrace(zap.ErrorLevel))
+
+	defer log.Sync()
```

golangci-lint (1.62.2)

Error return value of `log.Sync` is not checked

(errcheck)